### DISCUSSION TITLE
addDoubletScores default LSIMethod

### DISCUSSION THREAD

**Discussion by ammaralsheik** [GreenleafLab/ArchR#1760] [Questions / Documentation]
Not sure why the default LSIMethod for addDoubletScores is 1 or "tf-logidf" while the remainder of the pipeline uses  2 or "log(tf-idf)". As far as I understand log(tf-idf) is superior and is described as such in the tutorial. 

Additionally, I find very weird issues when I attempt to use it on most samples. Here is an example below of the same sample:
<img width="659" alt="Screen Shot 2022-12-01 at 10 32 36 AM" src="https://user-images.githubusercontent.com/7485410/205093692-10e86d0e-6c75-4f6e-8782-c643e5ba2f77.png">

**Comment by rcorces** [comment]
I'm not sure I have the satisfying answer you are looking for. Your observation is not true across every dataset and I'm not sure why we originally selected the default to be `LSIMethod = 1` (`tf-logidf`). But, in the case of the tutorial data, these two methods dont appear to give such different results

LSIMethod = 1 (`tf-logidf`)
![image](https://user-images.githubusercontent.com/10343046/205935536-5000b335-3569-439d-9c57-f5a274acb45d.png)

LSIMethod = 2 (`log(tf-idf)`)
![image](https://user-images.githubusercontent.com/10343046/205935657-50174d04-5cb5-48ee-af78-8a23a8359ac6.png)

One thing that I did notice in running the two different methods on the tutorial data is that `LSIMethod=2` (`log(tf-idf)`) gives some LSI dimensions that are highly correlated with `log10(depth+1)` and thus removed.

As to why you observe what you posted above, I'm not sure.

**Reply by ammaralsheik** [reply]
Hello @rcorces,

Thanks for your response, much appreciated; however, my concern is not only on the visuals of how the plots look but also at the resulting set of cells identified as doublets. As you can see below from the same sample from the tutorial data the doublet scores are not correlated necessarily. If this doubScores metric will be used for doublet filtering it will result in different cells being filtered.

Thoughts?

<img width="522" alt="Screen Shot 2022-12-08 at 4 54 57 PM" src="https://user-images.githubusercontent.com/7485410/206575574-41437bc9-13b3-4d65-af7a-7eab1998d360.png">

**Reply by ammaralsheik** [reply]
Here is the tutorial project showing the overlap of the remaining cells after doublet removal. Based on the number of cells in the tutorial 410 cells are filtered using the default filteratio. This keeps 10,250 cells but the overlap of the actual cells that remain following application of different LSIMethods for doublet identification have a significant difference only small proportion of the 410 cells that are filtered are the same cells.
<img width="552" alt="Screen Shot 2022-12-08 at 9 38 06 PM" src="https://user-images.githubusercontent.com/7485410/206611927-adf84bd3-2645-4482-90bb-6537f2b5e29c.png">

**Reply by rcorces** [reply]
Thanks for thoroughly going through this. I'll need some more time to look into this.

My first point is that we know that the doubScores are less consistent than doublet enrichment which is pointed out in the manual. That doesnt explain away the difference in called doublets that you are seeing though.

**Reply by ammaralsheik** [reply]
Of course, thank you all for an amazing package, hoping this can help clarify this important aspect.
I extracted the doubletEnrich scores as well for the tutorial data, using the two methods again, and you can see below, hope this helps. I appreciate the time you are taking to look into it.

<img width="888" alt="Screen Shot 2022-12-09 at 9 29 24 PM" src="https://user-images.githubusercontent.com/7485410/206824377-de7960ca-69d1-418a-8970-ded289c646cc.png">

**Reply by ammaralsheik** [reply]
Perhaps the preference for LSIMethod = 1 comes from this benchmarking below: (even though the formula in the figure looks different which is somewhat confusing log(TF)-IDF in the figure vs TF-log(IDF) from the package.)
https://www.nature.com/articles/s41588-021-00790-6/figures/7

**Reply by ammaralsheik** [reply]
@rcorces 
Upon further testing using internal dataset of PBMC samples with varying the cell input per reaction, we observed negative correlation between the number of cells per sample in Arch (cells passing filter) and the % of cells identified as potential doublets (doubletEnrich score >1 using default parameters of addDoubletScores).
Conceptually this should be the opposite as we expect higher percentage of doublets called when number of cells is higher for 10x applications. Unless there is something i'm not getting about the logic employed.
I do understand that these are doublet enrichment scores and not all of them will be filtered and that the filtration does take into account the number of cells in the sample and allows more doublet filtration. However, if the underlying identification is inversly correlated with number of cells then that persents a challenge.
I know that this is a larger number of cells than the assay is meant to process based on 10x # of cells guidance, nevertheless it's an interesting question and if doublet identification is robust across range of cell number inputs, that will be very useful.
I realize this might be a separate issue even though it is somewhat related, please let me know if I need to move it somewhere else.
<img width="492" alt="Screen Shot 2022-12-14 at 10 19 41 AM" src="https://user-images.githubusercontent.com/7485410/207635388-fa8d8f5f-cd07-4845-90d8-c68839aed170.png">
